underfs/orangefs/src/main/java/alluxio/underfs/orangefs/OrangeFSUnderFileSystem.java

```diff
@@ -0,0 +1,530 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.orangefs;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.retry.CountingRetry;
+import alluxio.retry.RetryPolicy;
+import alluxio.underfs.UnderFileSystem;
+
+import org.orangefs.usrint.Orange;
+import org.orangefs.usrint.OrangeFileSystemInputStream;
+import org.orangefs.usrint.OrangeFileSystemLayout;
+import org.orangefs.usrint.OrangeFileSystemOutputStream;
+import org.orangefs.usrint.PVFS2POSIXJNIFlags;
+import org.orangefs.usrint.PVFS2STDIOJNIFlags;
+import org.orangefs.usrint.Stat;
+import org.orangefs.usrint.Statfs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Stack;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * OrangeFS {@link UnderFileSystem} implementation using OrangeFS JNI shim and Direct Interface.
+ */
+@ThreadSafe
```

Comment:
Likely. Because OrangeFS client library can guarantee thread safe. In addition, the annotation of `@ThreadSafe` exists in other Alluxio underFS implementations. So I just keep it as it.